Classes/DDFileLogger.m

```diff
@@ -869,6 +874,32 @@ - (void)maybeRollLogFileDueToSize {
     }
 }
 
+- (void)maybeRollLogFileDueToDate {
+	if (_rollEveryDay && _currentLogFileInfo && ![self isLogFileCreatedToday:_currentLogFileInfo]) {
+		NSLogVerbose(@"DDFileLogger: Rolling log file due to date...");
+		
+		[self rollLogFileNow];
+	}
+}
+
+- (BOOL)isLogFileCreatedToday:(DDLogFileInfo *)aLogFile {
+	NSDate *logFileCreationDate = aLogFile.creationDate;
+	NSDate *today = [NSDate date];
```

Comment:
I think, problems will be only if user moving to different timezones due the day, but in that case time in logged messages will be strange too.

File logger class has no template methods, or another ways to override rolling without pain.
If I could, I've did that already. But to achieve same behavior I should rewrite private methods, htat not cool, because they could be changed someday. And my logger become broken.